src/abTestingInfra/Variant/index.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { assertNameIsValid } from '../utils';
+import ExperimentContext from '../ExperimentContext';
+import { VariantProps } from '../types';
+import { recordWin } from '../emitter';
+
+function Variant(props: VariantProps): JSX.Element {
+  const { name, children } = props;
+
+  assertNameIsValid(name);
+
+  if (typeof children === 'function') {
+    return (
+      <ExperimentContext.Consumer>
+        {(experimentName) => {
+          const winFunction = (): void => recordWin(experimentName, name);
+
+          return children(winFunction);
+        }}
+      </ExperimentContext.Consumer>
+    );
+  }
+
+  if (React.isValidElement(children)) {
```

Comment:
Idk if we can `import { isValidElement } from 'react';`, but if we can please do!

Reply:
done